gravitee-am-secrets/src/main/java/io/gravitee/am/service/secrets/evaluators/SecretsResolvingPluginConfigurationEvaluator.java

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.am.service.secrets.evaluators;
+
+import io.gravitee.am.plugins.handlers.api.core.PluginConfigurationEvaluator;
+import io.gravitee.am.service.secrets.resolver.SecretResolver;
+import io.gravitee.el.TemplateContext;
+import io.gravitee.el.TemplateEngine;
+import io.gravitee.secrets.api.annotation.Secret;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.util.ReflectionUtils;
+
+import java.lang.reflect.Field;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * @author GraviteeSource Team
+ */
+@Slf4j
+@RequiredArgsConstructor
+public class SecretsResolvingPluginConfigurationEvaluator implements PluginConfigurationEvaluator {
```

Comment:
Wondering if the `@ConfigurationEvaluator` from gravitee-plugin-annotation-processors would have also worked for AM?